util/types/src/core/cell.rs

```diff
@@ -625,6 +632,7 @@ pub fn resolve_transaction<CP: CellProvider, HC: HeaderChecker, S: BuildHasher>(
     seen_inputs: &mut HashSet<OutPoint, S>,
     cell_provider: &CP,
     header_checker: &HC,
+    no_immature_header_deps: bool,
```

Comment:
How about adding a bitsflags enum here?

```
TransactionResolvingOptions {
  Default = 0x0,
  ImmatureHeaderDeps = 0x1,
}

default => no flags
```

Then we can define two functions:

* `resolve_transaction`: resolve using the default options (no flags)
* `resolve_transaction_with_options`: add a new `TransactionResolvingOptions` parameter.

To resolve depending on the context, we can add a function which derive the correct TransactionResolvingOptions from the chain context.

Reply:
The function `resolve_transaction` without options is only used for tests.